THEMES-1195 Updated headline, overline and image opacity

malavikakoppula opened the pull request:
## Description

Review doc: https://paper.dropbox.com/doc/Themes-hover-states-audit--B400R8DxsjiVy0YImyMBZ0B4Ag-JyHvxHuZttlIfj5BzjTI3

Redesigned button components: https://www.figma.com/file/vWVGgZfZVBdSX7ACo3kmjr/Enhanced-Styling-Design-System?type=design&node-id=7736%3A104212&t=wkfbbYPc5dtqGGKM-1 - Connect your Figma account 

## Jira Ticket

- [THEMES-1195](https://arcpublishing.atlassian.net/browse/THEMES-1195)

## Acceptance Criteria

Applies to all promos (manual and non, S-XL):

1. image when link:

   a. opacity change to 80%

2. overline (only on XL, L):

    a. rest: $text-color

    b. hover: $text-color-subtle

3. headline: 

     a. rest: $text-color

     b. hover: $text-color-subtle

4. author name, when link (covered in linked ticket so no need to override but confirm as part of this ticket):

     a. rest: $text-color

     b. hover: $text-color-subtle

## Test Steps

1. Checkout this branch `git checkout THEMES-1195`
2. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/extra-large-manual-promo-block,@wpmedia/extra-large-promo-block,@wpmedia/large-manual-promo-block,@wpmedia/large-promo-block,@wpmedia/medium-manual-promo-block,@wpmedia/medium-promo-block,@wpmedia/small-manual-promo-block,@wpmedia/small-promo-block`
3. add all the blocks and inspect on hover to check the color

## Dependencies or Side Effects

- Feature pack pr for local development: https://github.com/WPMedia/arc-themes-feature-pack/pull/431
- Components pr. for local development: https://github.com/WPMedia/arc-themes-components/pull/199

## Author Checklist

_The author of the PR should fill out the following sections to ensure this PR is ready for review._

- [ ] Confirmed all the test steps a reviewer will follow above are working.
- [ ] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [ ] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [ ] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [ ] Confirmed this PR has unit test files
  - [ ] Ran `npm run test`, made sure all tests are passing
  - [ ] If the amount of work to write unit tests for this change are excessive,
        please explain why (so that we can fix it whenever it gets refactored).
- [ ] Confirmed relevant documentation has been updated/added.

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr.
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.


[THEMES-1195]: https://arcpublishing.atlassian.net/browse/THEMES-1195?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

Commit:
Updated headline, overline and image opacity

Commit:
Fixed image opacity

Commit:
removed console statements

Commit:
Merge branch 'arc-themes-release-version-2.0.3' into THEMES-1195

Commit:
Merge branch 'arc-themes-release-version-2.0.3' into THEMES-1195

malavikakoppula committed:
Merge branch 'arc-themes-release-version-2.0.3' into THEMES-1195

vgalatro commented on blocks/extra-large-manual-promo-block/_index.scss:
```diff
@@ -3,4 +3,20 @@
+			@include scss.block-properties("xl-manual-promo-headline-hover");
+			@include scss.block-components("xl-manual-promo-headline-hover");
+		}
+	}
```
We shouldn't need new tokens for the hover state specifically, just for overline and headline.

malavikakoppula replied:
@vgalatro Fixed this removing hover from here

vgalatro commented on blocks/extra-large-manual-promo-block/_index.scss:
```diff
@@ -3,4 +3,20 @@
+			@include scss.block-components("xl-manual-promo-headline-hover");
+		}
+	}
+	&__imgWithLink {
```
This class should target the image, whether or not it is linked. We can use the "components" styling to then target the link in the image block.

malavikakoppula replied:
@vgalatro Fixed this

vgalatro commented on blocks/extra-large-manual-promo-block/themes/news.json:
```diff
@@ -32,5 +38,29 @@
+			"desktop": {}
+		}
+	},
+	"xl-manual-promo-img-with-link": {
```
As noted above, this should be changed to always target the image.

malavikakoppula replied:
@vgalatro Fixed this by adding image component to this

vgalatro commented on blocks/extra-large-manual-promo-block/features/extra-large-manual-promo/default.jsx:
```diff
@@ -131,7 +135,10 @@ const ExtraLargeManualPromo = ({ customFields }) => {
 								>
-									<Image {...imageParams} />
+									<Image
+										className={`${linkURL ? `${BLOCK_CLASS_NAME}__imgWithLink` : ""}`}
```
Remove this conditional and always add a classname here, `${BLOCK_CLASS_NAME}__img`

malavikakoppula replied:
@vgalatro updated this

vgalatro commented on blocks/extra-large-manual-promo-block/themes/news.json:
```diff
@@ -32,5 +38,29 @@
+	"xl-manual-promo-img-with-link": {
+		"styles": {
+			"default": {
+				"opacity": 0.8
```
This should work as 
```
"components": {
  "link-hover": {
      "opacity": 0.8
    }
}
```

malavikakoppula replied:
@vgalatro Instead of link-hover, it's working with image

vgalatro reviewed (changes requested):
I made comments on the extra large manual promo, but they apply to all the promo blocks. We want to make sure we're using the "components" styling to keep things consistent and make sure inheritance is working correctly.

Commit:
Merge branch 'arc-themes-release-version-2.0.3' into THEMES-1195

Commit:
Merge branch 'arc-themes-release-version-2.0.3' into THEMES-1195

Commit:
Fixed hover state and images with link

Commit:
Merge branch 'arc-themes-release-version-2.0.3' into THEMES-1195

Commit:
chnaged imgWithLink to img

vgalatro commented on blocks/extra-large-manual-promo-block/_index.scss:
```diff
@@ -3,4 +3,16 @@
+	}
+	&__headline {
+		@include scss.block-properties("xl-manual-promo-headline-hover");
+		@include scss.block-components("xl-manual-promo-headline-hover");
```
Remove `-hover` from these token names.

vgalatro commented on blocks/extra-large-manual-promo-block/themes/news.json:
```diff
@@ -32,5 +38,39 @@
+		"styles": {
+			"default": {
+				"components": {
+					"heading": {
```
This should be `link-hover`, not `heading`. This is making regular text headings grey.

vgalatro commented on blocks/extra-large-manual-promo-block/themes/news.json:
```diff
@@ -32,5 +38,39 @@
+	"xl-manual-promo-img": {
+		"styles": {
+			"default": {
+				"link-hover": {
+					"opacity": 0.8
+				}
```
Needs to be wrapped in the `"components" : { }` property.

vgalatro replied:
In the `news.json` files for the other blocks the `"image"` property also needs to be changed to `"link-hover"`

vgalatro commented on blocks/extra-large-manual-promo-block/themes/news.json:
```diff
@@ -32,5 +38,39 @@
+			"desktop": {}
+		}
+	},
+	"xl-manual-promo-headline-hover": {
```
Make sure to remove `-hover` from the token name here too.

vgalatro replied:
Since we need to style the `link` that's a child of the `heading` component, we'll still need the `*-promo-headline` tokens for the blocks.

vgalatro commented on blocks/extra-large-manual-promo-block/themes/news.json:
```diff
@@ -32,5 +38,39 @@
+			},
+			"desktop": {}
+		}
+	},
```
The more I look at it, I don't think we need this token at all. Since all we need to do is style the component itself the `"overline-hover"` rule can be a part of `"xl-manual-promo"`.

vgalatro commented on blocks/extra-large-manual-promo-block/themes/news.json:
```diff
@@ -3,12 +3,18 @@
 					},
+					"overline": {
+						"color": "var(--text-color)"
+					},
```
To go with the other comment, we can move
```
"overline-hover": {
  "color": "var(--text-color-subtle)"
}
```
here, and eliminate the `*-promo-overline` tokens.

vgalatro commented on blocks/extra-large-manual-promo-block/features/extra-large-manual-promo/default.jsx:
```diff
@@ -96,12 +96,16 @@ const ExtraLargeManualPromo = ({ customFields }) => {
 			<article className={BLOCK_CLASS_NAME}>
-				{availableOverline ? <Overline href={overlineURL}>{availableOverline}</Overline> : null}
+				{availableOverline ? (
+					<Overline href={overlineURL} className={`${BLOCK_CLASS_NAME}__overline`}>
```
No longer need the `className` here.

vgalatro commented on blocks/extra-large-manual-promo-block/_index.scss:
```diff
@@ -3,4 +3,16 @@
 .b-xl-manual-promo {
 	@include scss.block-properties("xl-manual-promo");
 	@include scss.block-components("xl-manual-promo");
+	&__overline {
+		@include scss.block-properties("xl-manual-promo-overline-hover");
+		@include scss.block-components("xl-manual-promo-overline-hover");
+	}
```
This token may be removed.

vgalatro reviewed (changes requested):
This is looking better, but I still found some issues. I made notes on the extra large promo block, but they apply to all blocks. We should be able to eliminate the `*-promo-overline` tokens, cutting the complexity of the CSS we generate.

Commit:
removed hover tokens

Commit:
Merge branch 'arc-themes-release-version-2.0.3' into THEMES-1195

malavikakoppula commented:
> This is looking better, but I still found some issues. I made notes on the extra large promo block, but they apply to all blocks. We should be able to eliminate the `*-promo-overline` tokens, cutting the complexity of the CSS we generate.

@vgalatro, I removed all the hover tokens and adjusted styles according to links.

vgalatro commented on blocks/extra-large-manual-promo-block/themes/news.json:
```diff
@@ -28,49 +28,22 @@
-				}
-			},
-			"desktop": {}
-		}
```
Sorry if my comments caused some confusion, but we still need a token for the image container. The `opacity` property affects text as well, so the links are too light.

The change needed here was to add the components wrapper like so:
```
"xl-manual-promo-img": {
    "styles": {
        "default": {
            "components": {
	        "link-hover": {
		    "opacity": 0.8
		}
            }
        },
        "desktop": {}
    }
}
```

malavikakoppula replied:
@vgalatro, Fixed it in [a622ba2](https://github.com/WPMedia/arc-themes-blocks/pull/1690/commits/a622ba26042fa29d7ded26bda4f557df321862c0)

vgalatro commented on blocks/extra-large-manual-promo-block/themes/news.json:
```diff
@@ -28,49 +28,22 @@
 					},
+					"link-hover": {
+						"color": "var(--text-color-subtle)",
+						"opacity": 0.8
```
Adding opacity here is making the links too light of a color.

malavikakoppula replied:
@vgalatro, Fixed it in [8a050cf](https://github.com/WPMedia/arc-themes-blocks/pull/1690/commits/8a050cfab3b00863edf7f93c08104a862ad876fe)

vgalatro commented on blocks/extra-large-manual-promo-block/features/extra-large-manual-promo/default.jsx:
```diff
@@ -131,7 +127,6 @@ const ExtraLargeManualPromo = ({ customFields }) => {
 									component={Link}
 									condition={linkURL}
 									href={formatURL(linkURL)}
-									className={`${BLOCK_CLASS_NAME}__img`}
```
We'll still need this class.

malavikakoppula replied:
@vgalatro Fixed it in [a622ba2](https://github.com/WPMedia/arc-themes-blocks/pull/1690/commits/a622ba26042fa29d7ded26bda4f557df321862c0)

vgalatro commented on blocks/extra-large-manual-promo-block/_index.scss:
```diff
@@ -3,16 +3,4 @@
-	&__img {
-		@include scss.block-properties("xl-manual-promo-img");
-		@include scss.block-components("xl-manual-promo-img");
-	}
```
Keep the image token so we can apply the opacity to the correct element.

malavikakoppula replied:
@vgalatro Fixed it in [a622ba2](https://github.com/WPMedia/arc-themes-blocks/pull/1690/commits/a622ba26042fa29d7ded26bda4f557df321862c0)

vgalatro reviewed (changes requested):
I think there may have been some confusion when we were talking about the changes needed. We still need a token targeting the images in each promo block, with the opacity property set to the link-hover component there. The way it is now, all of the links change opacity on hover, making them the incorrect color.

Commit:
reverted img token and class

Commit:
Merge branch 'arc-themes-release-version-2.0.3' into THEMES-1195

Commit:
Removed opacity from link-hover

Commit:
removed opacity from link-hover

vgalatro commented on blocks/extra-large-promo-block/_index.scss:
```diff
@@ -1,6 +1,10 @@
 
 .b-xl-promo {
+	&__img {
+		@include scss.block-properties("xl-manual-promo-img");
+		@include scss.block-components("xl-manual-promo-img");
```
These should be `xl-promo-img`.

malavikakoppula replied:
@vgalatro Thanks, fixed it [808d50b](https://github.com/WPMedia/arc-themes-blocks/pull/1690/commits/808d50ba8aceca28b1da67cb8472f5949c731f92)

vgalatro commented on blocks/large-manual-promo-block/_index.scss:
```diff
@@ -1,6 +1,10 @@
 
 .b-large-manual-promo {
+	&__img {
+		@include scss.block-properties("xl-manual-promo-img");
+		@include scss.block-components("xl-manual-promo-img");
```
These should be `large-manual-promo-img`.

malavikakoppula replied:
@vgalatro fixed it [808d50b](https://github.com/WPMedia/arc-themes-blocks/pull/1690/commits/808d50ba8aceca28b1da67cb8472f5949c731f92)

vgalatro commented on blocks/large-promo-block/_index.scss:
```diff
@@ -1,6 +1,10 @@
 
 .b-large-promo {
+	&__img {
+		@include scss.block-properties("xl-manual-promo-img");
+		@include scss.block-components("xl-manual-promo-img");
```
These should be `large-promo-img`.

malavikakoppula replied:
@vgalatro fixed it [808d50b](https://github.com/WPMedia/arc-themes-blocks/pull/1690/commits/808d50ba8aceca28b1da67cb8472f5949c731f92)

vgalatro commented on blocks/medium-manual-promo-block/_index.scss:
```diff
@@ -1,6 +1,10 @@
 
 .b-medium-manual-promo {
+	&__img {
+		@include scss.block-properties("xl-manual-promo-img");
+		@include scss.block-components("xl-manual-promo-img");
```
These should be `medium-manual-promo-img`

malavikakoppula replied:
@vgalatro fixed it [808d50b](https://github.com/WPMedia/arc-themes-blocks/pull/1690/commits/808d50ba8aceca28b1da67cb8472f5949c731f92)

vgalatro commented on blocks/medium-promo-block/_index.scss:
```diff
@@ -1,6 +1,10 @@
 
 .b-medium-promo {
+	&__img {
+		@include scss.block-properties("xl-manual-promo-img");
+		@include scss.block-components("xl-manual-promo-img");
```
These should be `medium-promo-img`

malavikakoppula replied:
@vgalatro fixed it [808d50b](https://github.com/WPMedia/arc-themes-blocks/pull/1690/commits/808d50ba8aceca28b1da67cb8472f5949c731f92)

vgalatro commented on blocks/small-manual-promo-block/_index.scss:
```diff
@@ -1,6 +1,10 @@
 .b-small-manual-promo {
+	&__img {
+		@include scss.block-properties("xl-manual-promo-img");
+		@include scss.block-components("xl-manual-promo-img");
```
These should be `small-manual-promo-img`

vgalatro commented on blocks/small-promo-block/_index.scss:
```diff
@@ -1,6 +1,10 @@
 
 .b-small-promo {
+	&__img {
+		@include scss.block-properties("xl-manual-promo-img");
+		@include scss.block-components("xl-manual-promo-img");
```
These should be `small-promo-img`

vgalatro reviewed (changes requested):
Everything is working, however I noticed there seems to be copy/paste errors in the token definitions inside the scss files. Once those are fixed I can approve this PR.

Commit:
Updated token definitions

vgalatro reviewed (approved):
All the changes look good, approved!

pardistaherzadeh commented:
Looks good to me. Ready to be merged. @vgalatro

malavikakoppula committed:
Merge branch 'arc-themes-release-version-2.0.3' into THEMES-1195
